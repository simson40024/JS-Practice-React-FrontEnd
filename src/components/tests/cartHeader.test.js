import CartHeader from "../cart/cartHeader";
import { createRoot } from "react-dom/client";
import { act } from "@testing-library/react";

let container = null;
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  container.remove();
  container = null;
});

describe("CartHeader", () => {
  it("should create CartHeader", () => {
    act(() => {
      createRoot(container).render(<CartHeader />);
    });
    expect(container).toMatchSnapshot();
  });
});
