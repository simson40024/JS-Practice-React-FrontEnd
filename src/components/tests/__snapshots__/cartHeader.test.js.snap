// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartHeader should create CartHeader 1`] = `
<div>
  <div
    class="cart-header"
  >
    <div
      class="cart-product"
    >
      <p>
        Product
      </p>
    </div>
    <div
      class="cart-price"
    >
      <p>
        Price
      </p>
    </div>
    <div
      class="cart-quantity"
    >
      <p>
        Quantity
      </p>
    </div>
    <div
      class="cart-total"
    >
      <p>
        Total
      </p>
    </div>
    <div
      class="cart-button"
    />
  </div>
  <hr />
</div>
`;
